pkg/k8sinit/apply.go

```diff
@@ -67,34 +67,37 @@ func (s *launcherScope) applyPart(ctx context.Context, c *Configuration) error {
 	}
 
 	for _, item := range []struct {
-		configFile     string
-		restartService string
-		args           map[string]*string
+		configFile      string
+		restartServices []string
+		args            map[string]*string
 	}{
-		{configFile: "kube-apiserver", restartService: "kubelite", args: c.ExtraKubeAPIServerArgs},
-		{configFile: "kubelet", restartService: "kubelite", args: c.ExtraKubeletArgs},
-		{configFile: "kube-proxy", restartService: "kubelite", args: c.ExtraKubeProxyArgs},
-		{configFile: "kube-controller-manager", restartService: "kubelite", args: c.ExtraKubeControllerManagerArgs},
-		{configFile: "kube-scheduler", restartService: "kubelite", args: c.ExtraKubeSchedulerArgs},
-		{configFile: "kubelite-env", restartService: "kubelite", args: c.ExtraKubeliteEnv},
-		{configFile: "containerd", restartService: "containerd", args: c.ExtraContainerdArgs},
-		{configFile: "containerd-env", restartService: "containerd", args: c.ExtraContainerdEnv},
-		{configFile: "k8s-dqlite", restartService: "k8s-dqlite", args: c.ExtraDqliteArgs},
-		{configFile: "k8s-dqlite-env", restartService: "k8s-dqlite", args: c.ExtraDqliteEnv},
-		{configFile: "cluster-agent", restartService: "cluster-agent", args: c.ExtraMicroK8sClusterAgentArgs},
-		{configFile: "cluster-agent-env", restartService: "cluster-agent", args: c.ExtraMicroK8sClusterAgentEnv},
-		{configFile: "apiserver-proxy", restartService: "apiserver-proxy", args: c.ExtraMicroK8sAPIServerProxyArgs},
-		{configFile: "apiserver-proxy-env", restartService: "apiserver-proxy", args: c.ExtraMicroK8sAPIServerProxyEnv},
-		{configFile: "etcd", restartService: "etcd", args: c.ExtraEtcdArgs},
-		{configFile: "etcd-env", restartService: "etcd", args: c.ExtraEtcdEnv},
-		{configFile: "flanneld", restartService: "flanneld", args: c.ExtraFlanneldArgs},
-		{configFile: "flanneld-env", restartService: "flanneld", args: c.ExtraFlanneldEnv},
-		{configFile: "cni-env", restartService: "", args: c.ExtraCNIEnv},
+		{configFile: "kube-apiserver", restartServices: []string{"kubelite"}, args: c.ExtraKubeAPIServerArgs},
+		{configFile: "kubelet", restartServices: []string{"kubelite"}, args: c.ExtraKubeletArgs},
+		{configFile: "kube-proxy", restartServices: []string{"kubelite"}, args: c.ExtraKubeProxyArgs},
+		{configFile: "kube-controller-manager", restartServices: []string{"kubelite"}, args: c.ExtraKubeControllerManagerArgs},
+		{configFile: "kube-scheduler", restartServices: []string{"kubelite"}, args: c.ExtraKubeSchedulerArgs},
+		{configFile: "kubelite-env", restartServices: []string{"kubelite"}, args: c.ExtraKubeliteEnv},
```

Comment:
I do not think we have all these env files.

Reply:
these are argument files, not env vars (except for `kubelite-env`)

Reply:
True, we don't really read this at the moment, this is resolved in https://github.com/canonical/microk8s/pull/4135)